tools/funccount.py

```diff
@@ -57,18 +57,25 @@ def signal_ignore(signal, frame):
 bpf_text = """
 #include <uapi/linux/ptrace.h>
 
-struct key_t {
+#define MAP_SIZE 10000
+
+struct val_t {
     u64 ip;
+    u64 value;
 };
-BPF_HASH(counts, struct key_t);
+BPF_TABLE("array", int, struct val_t, counts, MAP_SIZE);
 
 int trace_count(struct pt_regs *ctx) {
     FILTER
-    struct key_t key = {};
-    u64 zero = 0, *val;
-    key.ip = ctx->ip;
-    val = counts.lookup_or_init(&key, &zero);
-    (*val)++;
+    struct val_t *val;
+    u64 zero = 0;
+    u32 key = ctx->ip % MAP_SIZE;
```

Comment:
Overkill is subjective. Is funccount.py meant to be fast, accurate, or safe? Seems that current implementation can't have all 3, so we need to choose. What is the purpose for funccount.py, and which of those 3 is the most appropriate tradeoff?